pkg/cache/tas_flavor_snapshot.go

```diff
@@ -379,26 +382,135 @@ func (s *TASFlavorSnapshot) Fits(flavorUsage workload.TASFlavorUsage) bool {
 // the TAS requests in the flavor handled by the snapshot.
 // The simulateEmpty parameter allows to look for the assignment under the
 // assumption that all TAS workloads are preempted.
-func (s *TASFlavorSnapshot) FindTopologyAssignmentsForFlavor(flavorTASRequests FlavorTASRequests, simulateEmpty bool) TASAssignmentsResult {
+func (s *TASFlavorSnapshot) FindTopologyAssignmentsForFlavor(flavorTASRequests FlavorTASRequests, simulateEmpty bool, wl *kueue.Workload) TASAssignmentsResult {
 	result := make(map[kueue.PodSetReference]tasPodSetAssignmentResult)
 	assumedUsage := make(map[utiltas.TopologyDomainID]resources.Requests)
 	for _, tr := range flavorTASRequests {
-		assignment, reason := s.findTopologyAssignment(tr, assumedUsage, simulateEmpty)
-		result[tr.PodSet.Name] = tasPodSetAssignmentResult{TopologyAssignment: assignment, FailureReason: reason}
-		if reason != "" {
-			return result
-		}
-		for _, domain := range assignment.Domains {
-			domainID := utiltas.DomainID(domain.Values)
-			if assumedUsage[domainID] == nil {
-				assumedUsage[domainID] = resources.Requests{}
+		if workload.HasFailedNodeAnnotation(wl) {
+			// In case of looking for replacement TopologyRequest has only
+			// PodSet with node to replace, so we match PodSetAssignment
+			psa := findPSA(wl, tr.PodSet.Name)
+			if psa == nil || psa.TopologyAssignment == nil {
+				continue
+			}
+
+			// We deepCopy the assignment, so if we delete unwanted domain, and
+			// There is no fit, we have the original assignment to retry with
+			assignment := psa.TopologyAssignment.DeepCopy()
+			nodeToReplace := wl.Annotations[kueuealpha.NodeToReplaceAnnotation]
+			tr.Count = deleteDomain(assignment, nodeToReplace)
+			if isStale, staleDomain := s.IsTopologyAssignmentStale(assignment); isStale {
+				result[tr.PodSet.Name] = tasPodSetAssignmentResult{TopologyAssignment: psa.TopologyAssignment, FailureReason: fmt.Sprintf("Domain %v not sound in the cache", staleDomain)}
```

Comment:
```suggestion
				result[tr.PodSet.Name] = tasPodSetAssignmentResult{TopologyAssignment: psa.TopologyAssignment, FailureReason: fmt.Sprintf("Domain %v not found in the cache", staleDomain)}
```
? 

Maybe we can also make it sound easier to understand to end user who may see it, maybe : "Cannot replace the <nodeToReplace> node, because the old assignment is invalid, as it contains the stale "<staleDomain>" domain / node.

Reply:
Done